src/themes/default/pages/Category.vue

```diff
@@ -86,9 +115,11 @@ export default {
         message: this.$t('Please select the field which You like to sort by'),
         action1: { label: this.$t('OK') }
       })
+    },
+    async changeFilter (filterVariant) {
+      this.$store.dispatch('category-magento/switchSearchFilter', filterVariant)
     }
-  },
-  mixins: [Category]
+  }
 }
 </script>
 
```

Comment:
I belive You should add the `beforeRouteEnter` here as well - to make sure it's called when the user clicks thru Breadcrumbs (staying at the Category page but switching to other categoires - probably asyncData won't be called in this case). The same with `beforeRouteUpdate` -> take a look at the td/vue-storefront/pull/2724/files#diff-eb024e62d2fc5209a823fa94c7ee0635L151

Reply:
sure, i'll test that afted adding breadcrumbs, but i'm 99% sure, that asyncData will be invoked